extensions/simple_extension.go

```diff
@@ -55,9 +56,9 @@ type TypeVariation struct {
 	Functions   TypeVariationFunctions
 }
 
-type Argument interface {
+type FuncParameter interface {
 	toTypeString() string
-	argumentMarker() // unexported marker method
```

Comment:
why remove the marker? We should keep to encapsulate the set of valid items.

Reply:
it doesn't accomplish the same. The unexported method guarantees that others can't implement additional implementations outside the package.

Reply:
Restored the marker method. 
Even without the marker method, this interface has one other unexported method(toTypeString).